mesonbuild/dependencies/python.py

```diff
@@ -111,21 +111,32 @@ def _check_version(self, version: str) -> bool:
     def sanity(self) -> bool:
         # Sanity check, we expect to have something that at least quacks in tune
 
-        import importlib.resources
-
-        with importlib.resources.path('mesonbuild.scripts', 'python_info.py') as f:
-            cmd = self.get_command() + [str(f)]
-            p, stdout, stderr = mesonlib.Popen_safe(cmd)
-
-        try:
-            info = json.loads(stdout)
-        except json.JSONDecodeError:
-            info = None
-            mlog.debug('Could not introspect Python (%s): exit code %d' % (str(p.args), p.returncode))
-            mlog.debug('Program stdout:\n')
-            mlog.debug(stdout)
-            mlog.debug('Program stderr:\n')
-            mlog.debug(stderr)
+        if self.path == sys.executable:
+            try:
+                from ..scripts import python_info
+                info = python_info.main()  # type: ignore[attr-defined]
+            except Exception:
+                import traceback
+                info = None
+                mlog.debug('Could not introspect internal Python')
+                mlog.debug('Error traceback:\n')
+                mlog.debug(traceback.format_exc())
+        else:
+            import importlib.resources
+            with importlib.resources.path('mesonbuild.scripts', 'python_info.py') as f:
+                cmd = self.get_command() + [os.fspath(f)]
```

Comment:
Here I've added the `os.fspath()`. It is needed because on Python 3.7 it is not possible to use a `pathlib.Path` as returned from `importlib.resources.path()` as a command array element for `subprocess.run()` and friends (which I believe bake the implementation of `mesonlib.Popen_safe()`). `mypy` complains if the `os.fspath()` is not present.